kyo-reactive-streams/shared/src/main/scala/kyo/interop/reactive-streams/StreamSubscription.scala

```diff
@@ -0,0 +1,173 @@
+                                StreamFinishState.StreamCanceled
+                            ))
+                        case Result.Panic(exception) => IO(throw exception).andThen(Loop.continue[
+                                StreamFinishState | Chunk[V],
+                                StreamFinishState,
+                                Async & Poll[Chunk[V]]
+                            ](StreamFinishState.StreamCanceled))
+                yield outcome
+            case state: StreamFinishState => Loop.done(state)
+        }
+    end poll
+
+    private[interop] def consume(
+        using
+        Tag[Emit[Chunk[V]]],
+        Tag[Poll[Chunk[V]]],
+        Frame,
+        Boundary[Ctx, IO & Abort[Nothing]]
+    ): Fiber[Nothing, StreamFinishState] < (IO & Ctx) =
+        Async
+            ._run[Nothing, StreamFinishState, Ctx](Poll.run(stream.emit)(poll).map(_._2))
+            .map { fiber =>
+                fiber.onComplete {
+                    case Result.Success(StreamFinishState.StreamComplete) => IO(subscriber.onComplete())
+                    case Result.Panic(e)                                  => IO(subscriber.onError(e))
+                    case _                                                => IO.unit
+                }.andThen(fiber)
+            }
+    end consume
+
+end StreamSubscription
+
+object StreamSubscription:
+
+    private[interop] enum StreamFinishState derives CanEqual:
+        case StreamComplete, StreamCanceled
+    end StreamFinishState
+
+    inline def subscribe[V, Ctx](
+        stream: Stream[V, Ctx],
+        subscriber: Subscriber[? >: V]
+    )(
+        using
+        Frame,
+        Tag[Emit[Chunk[V]]],
+        Tag[Poll[Chunk[V]]]
+    ): StreamSubscription[V, Ctx] < (IO & Ctx & Resource) =
+        _subscribe(stream, subscriber)
+
+    private[kyo] inline def _subscribe[V, Ctx](
+        stream: Stream[V, Ctx],
+        subscriber: Subscriber[? >: V]
+    )(
+        using
+        Frame,
+        Boundary[Ctx, IO & Abort[Nothing]],
+        Tag[Emit[Chunk[V]]],
+        Tag[Poll[Chunk[V]]]
+    ): StreamSubscription[V, Ctx] < (IO & Ctx & Resource) =
+        for
+            subscription <- IO.Unsafe(new StreamSubscription[V, Ctx](stream, subscriber))
+            _            <- subscription.subscribe
+            _            <- Resource.acquireRelease(subscription.consume)(_.interrupt.unit)
+        yield subscription
+
+    object Unsafe:
+        inline def subscribe[V, Ctx](
+            stream: Stream[V, Ctx],
+            subscriber: Subscriber[? >: V]
+        )(
+            subscribeCallback: (Fiber[Nothing, StreamFinishState] < (IO & Ctx)) => Unit
+        )(
+            using
+            AllowUnsafe,
+            Frame,
+            Tag[Emit[Chunk[V]]],
+            Tag[Poll[Chunk[V]]]
+        ): StreamSubscription[V, Ctx] =
+            _subscribe(stream, subscriber)(subscribeCallback)
+
+        private[kyo] inline def _subscribe[V, Ctx](
```

Comment:
Same trick with `Async._run`, now we don't have to pass context params explicitly